tests/IntegrationTests/WebAppCallsApiCallsGraphUiTests/TestingWebAppCallsApiCallsGraphLocally.cs

```diff
@@ -0,0 +1,135 @@
+        private const uint TodoListClientPort = 44321;
+        private const uint GrpcPort = 5001;
+        private const string SignOutPagePath = @"/MicrosoftIdentity/Account/SignedOut";
+        private const string TodoTitle1 = "Testing create todo item";
+        private const string TodoTitle2 = "Testing edit todo item";
+        //private const string ManagedIdentityObjectId = "9c5896db-a74a-4b1a-a259-74c5080a3a6a";
+        private string UiTestAssemblyLocation = typeof(TestingWebAppCallsApiCallsGraphLocally).Assembly.Location;
+        private readonly ITestOutputHelper _output;
+
+        public TestingWebAppCallsApiCallsGraphLocally(ITestOutputHelper output)
+        {
+            _output = output;
+            UiTestHelpers.InstallPlaywrightBrowser(); // will put in a fixture if this works
+        }
+
+        [Fact]
+        public async Task ChallengeUser_MicrosoftIdentityFlow_LocalApp_ValidEmailPasswordCreds_TodoAppFunctionsCorrectly()
+        {
+            // Arrange web app setup
+            Environment.SetEnvironmentVariable("ASPNETCORE_ENVIRONMENT", "Development");
+            //Environment.SetEnvironmentVariable("Azure_Client_Id", ManagedIdentityObjectId);
+            //Environment.SetEnvironmentVariable("AzureAd_ClientCredentials_0_ManagedIdentityClientId", ManagedIdentityObjectId);
+            Process? grpcProcess = UiTestHelpers.StartProcessLocally(UiTestAssemblyLocation, DevAppPath + GrpcPath, GrpcExecutable, GrpcPort);
+            Process? clientProcess = UiTestHelpers.StartProcessLocally(UiTestAssemblyLocation, DevAppPath + TodoListClientPath, TodoListClientExecutable, TodoListClientPort);
+            Process? serviceProcess = UiTestHelpers.StartProcessLocally(UiTestAssemblyLocation, DevAppPath + TodoListServicePath, TodoListServiceExecutable, TodoListServicePort, true);
+            
+            // Arrange Playwright setup, to see the browser UI, set Headless = false
+            using IPlaywright playwright = await Playwright.CreateAsync();
+            IBrowser browser = await playwright.Chromium.LaunchAsync(new() { Headless = true });
+            IBrowserContext context = await browser.NewContextAsync(new BrowserNewContextOptions { IgnoreHTTPSErrors = true });
+            await context.Tracing.StartAsync(new() { Screenshots = true, Snapshots = true, Sources = true });
+
+            try
+            {
+                if ( !UiTestHelpers.ProcessesAreAlive(new List<Process>() { clientProcess!, serviceProcess!, grpcProcess! }))
+                    {
+                        Assert.Fail($"Could not run web app locally.");
+                    }
+
+                IPage page = await context.NewPageAsync();
+                await page.GotoAsync(LocalhostUrl + TodoListClientPort);
+                LabResponse labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+
+                // Initial sign in
+                _output.WriteLine("Starting web app sign-in flow");

```

Comment:
super nit: add a ". " at the end of each for readabiltiy

Reply:
done